auacm/app/modules/competition_manager/views.py

```diff
@@ -31,6 +32,82 @@ def get_competitions():
     })
 
 
+@app.route('/api/competitions', methods=['POST'])
+@login_required
+def create_competition():
+    data = request.form
+    if current_user.admin == 0:
+        return serve_error('Only admins can create competitions', 401)
+    if not data['name'] or not data['start_time'] or not data['length'] or\
```

Comment:
From what I understand, it's more pythonic to [ask for forgiveness rather than permission](http://docs.quantifiedcode.com/python-code-patterns/readability/asking_for_permission_instead_of_forgiveness_when_working_with_files.html). (Yes, the link talks about files specifically, but the principle is the same.) In that sense, it may be better to put this in a try/except block an assume that the form has all the elements. (See problem_manager/views.py create_problem() method for an example)